platform/darwin/src/NSPredicate+MGLAdditions.h

```diff
@@ -1,9 +1,108 @@
 #import <Foundation/Foundation.h>
 
+#import "NSExpression+MGLAdditions.h"
 #include <mbgl/style/filter.hpp>
 
 @interface NSPredicate (MGLAdditions)
 
 - (mbgl::style::Filter)mgl_filter;
 
 @end
+
+namespace mbgl {
+namespace darwin {
+
+class FilterEvaluator {
+public:
+
+mbgl::style::Filter operator()(NSComparisonPredicate *predicate) {
+    switch (predicate.predicateOperatorType) {
```

Comment:
The entire FilterEvaluator definition shouldn't be inlined in the header. Declare the operators here but define them in the implementation file.